book/tables/pharmacokinetic/pkpt03.qmd

```diff
@@ -17,23 +17,80 @@ library(scda)
 library(dplyr)
 library(tern)
 
+# Preprocess analysis data ----
 adpp <- synthetic_cdisc_dataset("latest", "adpp")
-adpp <- adpp %>% filter(PPSPEC == "Plasma", AVISIT == "CYCLE 1 DAY 1")
+adpp <- adpp %>%
+  filter(PPSPEC == "Plasma") %>%
+  filter(AVISIT %in% c("CYCLE 1 DAY 1", "CYCLE 1 DAY 2")) %>%
+  h_pkparam_sort() %>%
+  mutate(PARAM = factor(paste0(TLG_DISPLAY, " (", AVALU, ")"))) %>%
+  mutate(PARAM = reorder(PARAM, TLG_ORDER))
+
+# Preprocess subject-level data ----
+adsl <- synthetic_cdisc_dataset("latest", "adsl")
+
+# Workaround needed to include (N=xx) population counts
+# Repeat ADSL by the number of levels in AVISIT
+# Include AVISIT and dummy PARAM as it's needed for trim_levels_in_group
+adsl_tmp <- adsl %>%
+  select(STUDYID, USUBJID, ARMCD) %>%
+  unique() %>%
+  mutate(PARAM = factor(NA_character_, levels = levels(adpp$PARAM)))
+
+# Data for Plasma Drug X example ----
+adpp_x <- adpp %>%
+  filter(PPCAT == "Plasma Drug X") %>%
+  # Please do not replicate mutate statement below!
+  # It is used to make the random data in this example more realistic
+  # as not all parameters are always available across all visits.
+  mutate(
+    AVAL = if_else(
+      ARMCD == "ARM A" & AVISIT == "CYCLE 1 DAY 1" & PARAM == "Cmax (ug/mL)",
+      NA_real_, AVAL
+    )
+  )
+
+adpp_x_tmp <- adpp_x %>%
+  select(STUDYID, USUBJID, ARMCD, AVISIT) %>%
+  unique()
+
+adsl_x_splitvars <- adsl_tmp %>%
+  left_join(adpp_x_tmp, by = c("STUDYID", "USUBJID", "ARMCD")) %>%
+  filter(!is.na(AVISIT))
+
+# Data for Plasma Drug Y example ----
+adpp_y <- adpp %>%
+  filter(PPCAT == "Plasma Drug Y")
+
+adpp_y_tmp <- adpp_y %>%
+  select(STUDYID, USUBJID, ARMCD, AVISIT) %>%
+  unique()
+
+adsl_y_splitvars <- adsl_tmp %>%
+  left_join(adpp_y_tmp, by = c("STUDYID", "USUBJID", "ARMCD")) %>%
+  filter(!is.na(AVISIT))
 ```
 
 ## Standard Table
 
 ```{r}
 # lyt creation
 lyt <- basic_table() %>%
+  split_rows_by(
+    var = "AVISIT",
+    split_fun = drop_split_levels,
```

Comment:
```suggestion
    split_fun = drop_split_levels,
    label_pos = "topleft",
```

Reply:
Let's wait until Qi confirms whether AVISIT should be a page-by variable. If so, then we shouldn't add the label to `topleft` because it will be repeated.

Reply:
@anajens Can you please remove the VISIT from subtitle, and keep VISIT in the table group-by varaible? Then both Cycle 1 Day 1 and Cycle 1 Day 2 will be shown in one table. Did I answer your question?

Reply:
Resolved offline. With `split_rows_by(..., page_by = TRUE)` it's not necessary to use `label_pos = "topleft"`.

Also I confirmed that no more changes are needed to the layout since `label_pos = "topleft"` was only a suggestion and not part of the code. So will go ahead and merge.